osf/management/commands/create_EGAP_json.py

```diff
@@ -0,0 +1,251 @@
+    registration['q36'] = build_nested_response('Agree')
+    make_json_file(filepath, registration, 'registration')
+
+def make_json_file(filepath, data, json_type):
+    if json_type == 'project':
+        filepath = filepath + '/project.json'
+    if json_type == 'registration':
+        filepath = filepath + '/registration-schema.json'
+    with open(filepath, 'w') as outfile:
+        json.dump(data, outfile)
+
+def build_question_response(header_row, row, question_key, column_title):
+    """Format the question's response to go in the registration_metadata
+    :param header_row: Header row in spreadsheet
+    :param row: Row in spreadsheet
+    :param question_key: string, Official question key as part of schema
+    :param column_title: string, Corresponding question_key column title in EGAP spreadsheet
+    """
+    index = header_row.index(column_title)
+    value = clean_value(row[index])
+    # Spreadsheet has these as comma-separated values, but looking for array
+    if question_key in ['q33', 'q34']:
+        value = value.split(', ')
+    return build_nested_response(value)
+
+def clean_value(value):
+    """Clean spreadsheet values of issues that will affect validation """
+    if value == 'n/a':
+        return 'N/A'
+    elif value == 'Design was registered before field was added':
+        return ''
+    return value
+
+def build_nested_response(value):
+    return {
+        'comments': [],
+        'extra': [],
+        'value': value
+    }
+
+def validate_response(qid, value):
+    """Validate question response
+
+    Validating each question response individually.  If there is an error, we will
+    attempt to add the value to the corresponding "Other" block.  Return that question id instead.
+
+    For example, q6 is a multiple choice question, with "Other" as a choice.  If text is entered
+    for q6 that does not match one of the multiple choice answers, assuming that this is "other"
+    text, and this response should go to the corresponding q7 question.  q6 will be marked
+    as "Other"
+
+    :param qid: string, question id from schema
+    :param value: question response
+    :param draft: DraftRegistration
+    :return qid: tuple, (qid corresponding to value, optional "Other" qid)
+    """
+    temporary_check = {}
+    temporary_check[qid] = value
+
+    try:
+        egap_schema = RegistrationSchema.objects.get(name='EGAP Registration')
+    except ObjectDoesNotExist:
+        # The proper EGAP Registration schema is not yet imported, need to throw error.
+        raise RuntimeError('Configuration error, run add_egap_registration_schema.py, then try again')
+
+    try:
+        egap_schema.validate_metadata(temporary_check)
+    except ValidationError as exc:
+        if qid in other_mapping:
+            return other_mapping[qid], qid
+        else:
+            raise Exception(exc)
+    return qid, None
+
+@celery_app.task(name='management.commands.create_EGAP_json')
```

Comment:
Could this be removed? I'm not sure if I can think of a good case for running this periodically.